app.js

```diff
@@ -0,0 +1,58 @@
+var createError = require('http-errors');
+var express = require('express');
+var path = require('path');
+var cookieParser = require('cookie-parser');
+var logger = require('morgan');
+var hbs = require('hbs');
+var fs = require('fs');
+var indexRouter = require('./routes/index');
+var usersRouter = require('./routes/users');
+
+var app = express();
+
+/*  
+ *  Section to register hbs partials according to the files 
+ *  present under the directory structure /views/partials
+ */
+var partialsDir = __dirname + '/views/partials';
+var filenames = fs.readdirSync(partialsDir);
+filenames.forEach(function (filename) {
+  var matches = /^([^.]+).hbs$/.exec(filename);
+  if (!matches) {
+    return;
```

Comment:
You should avoid inconsistent returns. Better check if match is not null, then register hbs partial.